documents/privacy.md

```diff
@@ -0,0 +1,301 @@
+
+We are in the process of tuning the cost for Scrypt, where we need to balance between:
+
+-   Providing as much protection as possible against brute force attacks
+
+-   CPU usage (and therefore battery consumption) on end-users\' phones - particularly considering that for reasons of inclusivity
+    (and because of some of the poorer countries we are targeting) we need the App to run even on low-end and old phones.
+    
+-   Total time & cost to compute hashes when generating "points of concern" data sets in our Safe Places solution for Contact Tracers.
+
+By computing hashes at the point that we record a location point on a user's smartphone, we spread out the cost of computing the hashes, avoiding the need to compute a large number of hashes when it comes to comparing location data with points of concern.
+
+The length of the hashes we output is 64 bits / 8 bytes.
+
+The total entropy, across the entire land surface of the earth, and 1 year of 5 minute time windows, is about 5 x 10\^16, or 58 bits. So 64 bits is plenty to make collisions unlikely over a couple of weeks, in the area covered by a single Health Authority.
+
+### Salts
+
+For the scrypt hash, we use a single, fixed salt.
+
+We are well aware that best practice with scrypt would be to use a long unique salt for each hash. However we have no way to co-ordinate the choice of salt between the server and the app.
+
+An alternative design that we considered was for the Health Authority to generate its hashes with a unique salt every time they are published.
+
+Unfortunately, this makes it impossible to pre-compute the hashes in the mobile app. In order to make it feasible for the app to compute the hashes at the point of comparison, it becomes necessary to reduce the cost of the hash function, which eliminates all the benefits of having a variable salt.
+
+We have also considered a variable global salt (e.g one for each calendar day of data), which is not known in advance. Without a secure channel to distribute the salt to the App, the salt will become public, and therefore be equally available to an attacker. Therefore the only significant benefit will be that it will prevent pre-computation of tables. This is a modest additional protection that we are considering as a future improvement - but we may choose to focus on more substantial improvements, as described below.
+
+### Code Obfuscation
+
+At this point in time, we have not attempted to obfuscate any of the code, or the value of the salt that we use.
+
+We are well aware that code obfuscation can easily be reversed (or that our open source obfuscated code could simply be lifted and used to encode hashes, without even de-obfuscating it).
+
+If we had a secure means of concealing the salty that is used, that would add considerably to the security of the published hashes, but we do not believe this is possible.
```

Comment:
If we had a secure means of concealing the salty that is used,   **is salty** correct?